### PR TITLE
Remove Noah-WRFv4, bug fix in Thompson MP inner loop, update HWRF regression tests, set ECF_TRIES to 2, update Cheyenne job submission scripts; contains "Reduce memory required by MERRA2 option; fix diag_tables for P7 tests; adds C384 P7 tests to Cheyenne.intel" (#866), contains "Update CMEPS" (#775)

### LINKED ISSUE #57: update to ESCOMP/master
1) The Xgrid implementation for calculating A-O fluxes only has been added for CESM at ESCOMP/master (ESCOMP PR [#216](https://github.com/ESCOMP/CMEPS/pull/216)). This capability should be brought to NOAA-EMC to allow development of the Xgrid for ufs-weather.

2) Accumulation field bundles have been refactored (ESCOMP PR [#226](https://github.com/ESCOMP/CMEPS/pull/226)). The import and export accumulation FBs are no longer arrays (carried for multiple components). The import accumulation FBs (not used in UFS) are now module specific (runoff and glacial ice). The ocean export accumulation FBs (used for the ocean export in UFS) is now carried in a specific ``FBExpAccumOcn``

3) Clean up of ``med.F90``.  When setting up the mediator mesh and related import/export fields from components on both grids and meshes, ``Med.F90`` utilizes an ``IPDv03p4 `` phase to modify the the decomp/distribution of the transferred Grid/Mesh.  In the refactored code, the process by which an incoming grid or mesh is modified (redistributed) to the mediator PEs is much more clear:  a new grid/mesh is created on the mediator PEs and upon leaving the phase, the coordinates and geom base of the old grid or mesh is automatically transferred to the new grid/mesh. 

    The transfer of fields from the incoming grid/mesh is completed in the subsequent ``IPDv03p5`` phase. For a transferred grid, the ``IPDv03p5`` creates the mesh and then transfers all fields in the State to the mesh in a ``NUOPC_realize`` call. This call acts to kick out the old Grid-based field from the state, and replace it with a Mesh-based one.

    It should be noted that the cleanup of med.F90 resolves an issue where the call to ``ESMF_FieldEmptyComplete`` resulted in a non-fatal error. Currently, an error message in the PET logs indicates an attempt to complete a field(s) that is already destroyed:
```
ESMF_Field.F90:314 ESMF_FieldValidate Invalid object  - Uninitialized or already destroyed Field: fieldstatus not ready
```
  This error did not cause the code to abort because the complete call was also missing a subsequent ``ChkErr`` call. The error was produced because for a transferred Grid, the call to ``NUOPC_realize`` had already replaced the grid fields with 
the mesh fields. Thus the incoming grid field had already been destroyed (replaced by the mesh field).

4) Mediator history refactor. Currently to obtain a coupler history file (e.g. debugging the coupled model), a special ``med_phase_history_write`` needs to be added within the run sequence. The coupler history file is also not produced at the right frequency when there are different slow and fast coupling frequency in the run sequence. The refactor of the coupler history writing will allow each component to produce a coupler history file separately; the frequency of the coupler history files will be a configurable option at run time, and can be set differently for each coupled component where a history file is requested.

This update requires ESMF8.2.0bs14 or higher. Additional settings will also need to be added to the nems.configure. 

This update will also change baselines because the mediator restart file will not contain the same fields due to the changes noted in item 2 above. Testing has shown that all common fields between the two mediator restarts and all other component model restart files are b4b with existing baselines.

### LINKED ISSUE #698: C384 P7 coupled tests fail on cheyenne
## Description

This job fails to run at startup with an error: ``MPT: shepherd terminated: r5i4n4.ib0.cheyenne.ucar.edu - job aborting``

### To Reproduce:

Cheyenne.intel

## Additional context

Found during testing for PR #639. The test was turned off for cheyenne.intel until the issue can be resolved. 

Since the wave model cannot be compile in debug mode, an equivalent test was created for a non-wave bmark_p7b configuration. The test can be accessed using rt.test in this [branch](https://github.com/DeniseWorthen/ufs-weather-model/tree/test2)

When running in debug mode, the model fails with the following:

```
2:MPT:     header=header@entry=0x7ffdcfe45c10 "MPT ERROR: Rank 32(g:32) received signal SIGFPE(8).\n\tProcess ID: 62280, Host: r7i7n1, Program: /glade/scratch/worthen/FV3_RT/rt_13027/cpld_bmark_v16_p7b/fv3.exe\n\tMPT Version: HPE MPT 2.22  03/31/20 15"...) at sig.c:340
32:MPT: #3  0x00002b177c55e4ff in first_arriver_handler (signo=signo@entry=8,
32:MPT:     stack_trace_sem=stack_trace_sem@entry=0x2b1786c00080) at sig.c:489
32:MPT: #4  0x00002b177c55e793 in slave_sig_handler (signo=8, siginfo=<optimized out>,
32:MPT:     extra=<optimized out>) at sig.c:565
32:MPT: #5  <signal handler called>
32:MPT: #6  0x0000000008047c67 in module_sf_noahmplsm::energy (parameters=..., ice=0,
32:MPT:     vegtyp=17, ist=1, nsnow=3, nsoil=4, isnow=0, dt=300,
32:MPT:     rhoair=1.1787577636261917, sfcprs=101003.09822185335,
32:MPT:     qair=0.011852218429786931, sfctmp=296.38129134030879,
32:MPT:     thair=296.38129134030879, lwdn=345.52102043841046, uu=3.4295500034750659,
32:MPT:     vv=-3.4672281739002062, zref=10.812501434103201,
32:MPT:     co2air=39.896223797632075, o2air=21109.64752836735, solad=..., solai=...,
32:MPT:     cosz=-0.91289500682162039, igs=1, eair=1910.8519305175637,
32:MPT:     tbot=299.26776123046875, zsnso=..., zsoil=..., elai=0, esai=0, fwet=0,
32:MPT:     foln=1, fveg=0, pahv=0, pahg=0, pahb=0, qsnow=0, dzsnso=...,
32:MPT:     lat=0.21661703129464016, canliq=0, canice=0, iloc=6, jloc=-9999,
32:MPT:     z0wrf=nan(0x7baddadbaddad), imelt=..., snicev=..., snliqv=..., epore=...,
32:MPT:     t2m=nan(0x7baddadbaddad), fsno=0, sav=0, sag=0,
32:MPT:     qmelt=nan(0x7baddadbaddad), fsa=0, fsr=0, taux=nan(0x7baddadbaddad),
32:MPT:     tauy=nan(0x7baddadbaddad), fira=nan(0x7baddadbaddad),
32:MPT:     fsh=nan(0x7baddadbaddad), fcev=nan(0x7baddadbaddad),
32:MPT:     fgev=nan(0x7baddadbaddad), fctr=nan(0x7baddadbaddad),
32:MPT:     trad=nan(0x7baddadbaddad), psn=nan(0x7baddadbaddad),
32:MPT:     apar=nan(0x7baddadbaddad), ssoil=nan(0x7baddadbaddad), btrani=...,
32:MPT:     btran=9.9999999999999995e-07, ponding=nan(0x7baddadbaddad),
32:MPT:     ts=nan(0x7baddadbaddad), latheav=nan(0x7baddadbaddad),
32:MPT:     latheag=nan(0x7baddadbaddad), frozen_canopy=3435973836,
32:MPT:     frozen_ground=3435973836, tv=291.71072387695312, tg=291.71072387695312,
32:MPT:     stc=..., snowh=0, eah=2000, tah=291.71072387695312, sneqvo=0, sneqv=0,
32:MPT:     sh2o=..., smc=..., snice=..., snliq=..., albold=0.65000000000000002, cm=0,
32:MPT:     ch=0, dx=-9999, dz8w=-9999, q2=0.011852218429786931, tauss=0, laisun=0,
32:MPT:     laisha=0, rb=0, errmsg=..., errflg=0, qc=-9999, qsfc=9.99e+20,
32:MPT:     psfc=101128.33749723693, t2mv=0, t2mb=nan(0x7baddadbaddad), fsrv=0,
32:MPT:     fsrg=0, rssun=nan(0x7baddadbaddad), rssha=nan(0x7baddadbaddad), albd=...,
32:MPT:     albi=..., albsnd=..., albsni=..., bgap=0, wgap=0,
32:MPT:     tgv=nan(0x7baddadbaddad), tgb=nan(0x7baddadbaddad),
32:MPT:     q1=nan(0x7baddadbaddad), q2v=0, q2b=nan(0x7baddadbaddad),
32:MPT:     q2e=nan(0x7baddadbaddad), chv=0, chb=nan(0x7baddadbaddad),
32:MPT:     emissi=nan(0x7baddadbaddad), pah=0, shg=0, shc=0,
32:MPT:     shb=nan(0x7baddadbaddad), evg=0, evb=nan(0x7baddadbaddad), ghv=0,
32:MPT:     ghb=nan(0x7baddadbaddad), irg=0, irc=0, irb=nan(0x7baddadbaddad), tr=0,
32:MPT:     evc=0, chleaf=0, chuc=0, chv2=0, chb2=nan(0x7baddadbaddad),
32:MPT:     .tmp.ERRMSG.len_V$698=512)
```

### LINKED ISSUE #865: diag_tables for P7 tests do not include merra2 variables
## Description

MERRA2 is used by default in the P7 tests (standalone and coupled). During the commit for PR #765 , the diag_tables used by the tests were not updated to include the MERRA2 output.

### To Reproduce:

Examine the forecast files for any P7 test. The MERRA2 variables (e.g, ``AOD_550``) are not present.

### PR DESCRIPTION
**Pull request by climbfuji**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Description

This PR removes unused/broken HAFS-HWRF regression tests that used Ferrier-Aligo MP, and updates the old HAFS-HWRF regression tests to use Noah MP instead of Noah-WRFv4 as instructed by @yangfanglin. It is understood that these old HAFS-HWRF regression tests will eventually be replaced with new HAFS regression tests.

The submodule pointers for fv3atm and ccpp-physics are updated to remove Noah-WRFv4 from fv3atm/ccpp-physics and to fix a bug in Thompson MP when the inner loop logic is used.

The changes in this PR and associated PRs require new baselines for the following four tests:
```
fv3_HAFS_v0_hwrf_thompson_debug
fv3_HAFS_v0_hwrf_thompson
fv3_esg_HAFS_v0_hwrf_thompson_debug
fv3_esg_HAFS_v0_hwrf_thompson
```
All other regression tests are b4b reproducible against the current baselines (tested on Hera with Intel and GNU). Note in particular that the Thompson MP results do not change, because the inner loop logic is not used in the regression tests.

The PR also contains an update from @DusanJovic-NOAA that uses the automatic rerun capability for ecflow: `ECF_TRIES` is set to 2 in this PR, which will attempt to rerun a failed test once before reporting a failure. This is to avoid unnecessary reruns of the entire regression test suite if single tests timed out.

Further, this PR as of 2021/10/28 contains the following changes:

- #866 (Reduce memory required by MERRA2 option; fix diag_tables for P7 tests; adds C384 P7 tests to Cheyenne.intel) from @DeniseWorthen. These updates require new baselines for several of the P7 tests, the details are described in #866: Bit-for-bit comparisons fail only because of the changes to the forecast files associated with the updated `diag_table`. All common fields in the forecast files are B4B and all restart files are B4B. This has been verified before including #775 (see next bullet).

- #775 (Update CMEPS) from @DeniseWorthen. These updates require new baselines for all coupled tests.

- Update Cheyenne job submission scripts due to a recent change on Cheyenne w.r.t. how the user environment gets propagated to the jobs: replace `#!/bin/sh` with `#/bin/bash` to fix random compile errors (`module command not found), as per recommendation by the CISL helpdesk. This change does not affect the results on Cheyenne.

No new input data required.

### Issue(s) addressed

- fixes https://github.com/NCAR/ccpp-physics/issues/734
- fixes #865 
- fixes #698
- fixes https://github.com/NOAA-EMC/CMEPS/issues/57

## Testing

Tested on Hera/Intel on 2021/10/25 against official baseline, only the four tests mentioned above failed with b4b differences.

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [X] CI - https://github.com/ufs-community/ufs-weather-model/pull/831/commits/f55e550e39106c40b36213d0055ef30b0599cb23

## Dependencies

- waiting on https://github.com/NCAR/ccpp-physics/pull/743
- waiting on https://github.com/NOAA-EMC/fv3atm/pull/398

### CODE REVIEW

File: tests/rt.conf
Comment (junwang-noaa) [inline tests/rt.conf:80]: I am curious, do we still need to compile with FV3_HAFS_v0_hwrf since the test is not run?

Comment (climbfuji) [reply tests/rt.conf:80]: See the change in line 66 - we don't compile the suite anymore, and this PR removes it. We should note, however, that this is the last suite that contains Ferrier-Aligo MP, which means that it is no longer tested to compile (we didn't test it to run correctly for a long time, because it doesn't). That may be ok, but we should be aware of it.

**Comment by DeniseWorthen** [comment]
I had a compile failure for the ``hafs_regional_atm`` and ``hafs_regional_atm_ocn`` test on cray because this PR requires esmf8.2bs14. The update to 8.2bs14 was done in PR #776 but it looks like the module list was not updated on cray. This PR adds a new NUOPC call in CMEPS which now fails using the older 811 esmf.

I added the library in the ufs_wcoss_cray

```
diff --git a/modulefiles/ufs_wcoss_cray b/modulefiles/ufs_wcoss_cray
index 01ca6e76..9586ca24 100644
--- a/modulefiles/ufs_wcoss_cray
+++ b/modulefiles/ufs_wcoss_cray
@@ -63,7 +63,7 @@ module load gni-headers
 module load udreg
 module load ugni

-module load esmf/811
+module load esmf/820bs14
 module load fms/2021.03

 module swap pmi pmi/5.0.11
```

In the compile_001/err I see as the first occurence of ``ESMF``:

```
+ export ESMFMKFILE=/gpfs/hps/usrx/local/nceplibs/NCEPLIBS/cmake/install/NCEPLIBS-v1.3.0/esmf/esmf-820bs14/lib/esmf.mk
```

But then:

```
CMake Error at /gpfs/hps3/emc/nems/noscrub/emc.nemspara/soft/cmake/cmake-3.20.1-linux-x86_64/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find ESMF (missing: ESMF_LIBRARY_LOCATION) (found version
  "8.2.0")
```

**Comment by junwang-noaa** [comment]
Denise, thanks for catching the wrong esmf lib on cray. Let's check with
nceplibs team if esmf820bs14 is installed on cray

**Comment by climbfuji** [comment]
> Denise, thanks for catching the wrong esmf lib on cray. Let's check with nceplibs team if esmf820bs14 is installed on cray
> […](#)

Yes, thanks Denise, also for running the wcoss tests in general. After cray is done, we are ready to merge. That was a quick one then, despite creating new baselines.

**Comment by DeniseWorthen** [comment]
Just started the cray verify step.

**Comment by DeniseWorthen** [comment]
CMEPS has been merged; hash 5beead0

**Comment by climbfuji** [comment]
> CMEPS has been merged; hash 5beead0

CMEPS submodule pointer updated, waiting for fv3atm to be merged

File: .gitmodules
Comment (climbfuji) [inline .gitmodules]: Should I revert this change? @DeniseWorthen

Comment (DusanJovic-NOAA) [inline .gitmodules]: Remove `.git`

**Comment by climbfuji** [comment]
fv3atm has is correct (4214e6d), waiting to resolve a .gitmodules change question. Please all review carefully once more, many unrelated changes.

**Review by DusanJovic-NOAA — APPROVED** [review]

File: .gitmodules
Comment (DeniseWorthen) [reply .gitmodules]: I saw that and was going to comment. Yes please.

**Comment by DeniseWorthen** [comment]
Double checked again. All my changes are present as expected. Thanks for pulling all these in together.

**Review by junwang-noaa — APPROVED** [review]
Thanks for including the ECF_TRIES update, hope it will resolve RT test problem caused by hera slurm issue.

**Comment by climbfuji** [comment]
Should be good now!

**Review by BinLiu-NOAA — APPROVED** [review]
Looks good to me.

**Review by DeniseWorthen — APPROVED** [review]

**Pull request merged**
